musmart/melody/contour/step_contour.py

```diff
@@ -0,0 +1,217 @@
+
+    _step_contour_length = 64
+
+    def __init__(
+        self,
+        pitches: list[int],
+        durations: list[float],
+        step_contour_length: int = _step_contour_length
+    ):
+        """Initialize StepContour with melody data.
+
+        Parameters
+        ----------
+        pitches : list[int]
+            List of pitch values
+        durations : list[float]
+            List of duration values measured in tatums
+        step_contour_length : int, optional
+            Length of the output step contour vector (default is 64)
+
+        References
+        ----------
+        [1] Müllensiefen, D. (2009). Fantastic: Feature ANalysis Technology Accessing
+        STatistics (In a Corpus): Technical Report v1.5
+
+        Examples
+        --------
+        >>> sc = StepContour([60, 62], [1.0, 1.0], step_contour_length=4)
+        >>> sc.contour
+        [60, 60, 62, 62]
+        """
+        if len(pitches) != len(durations):
+            raise ValueError(
+                f"The length of pitches (currently {len(pitches)}) must be equal to "
+                f"the length of durations (currently {len(durations)})"
+            )
+
+        self._step_contour_length = step_contour_length
+        self.contour = self._calculate_contour(pitches, durations)
+
+    def _normalize_durations(self, durations: list[float]) -> list[float]:
+        """Helper function to normalize note durations to fit within 4 bars of 4/4 time
+        (64 tatums total).
+
+        Parameters
+        ----------
+        durations : list[float]
+            List of duration values measured in tatums
+
+        Returns
+        -------
+        list[float]
+            List of normalized duration values
+
+        Examples
+        --------
+        >>> sc = StepContour([60], [1.0])
+        >>> sc._normalize_durations([2.0, 2.0])
+        [32.0, 32.0]
+        """
+        total_duration = sum(durations)
+        if total_duration == 0:
+            return durations
+
+        normalized = [
+            self._step_contour_length * (duration / total_duration)
+            for duration in durations
+        ]
+        return normalized
+
+    def _expand_to_vector(
+        self,
+        pitches: list[int],
+        normalized_durations: list[float]
+    ) -> list[int]:
+        """Helper function to create a vector of length step_contour_length by repeating
+        each pitch value proportionally to its normalized duration.
+
+        Parameters
+        ----------
+        pitches : list[int]
+            List of pitch values
+        normalized_durations : list[float]
+            List of normalized duration values (should sum to step_contour_length)
+
+        Returns
+        -------
+        list[int]
+            List of length step_contour_length containing repeated pitch values
+
+        Examples
+        --------
+        >>> sc = StepContour([60], [1.0], step_contour_length=4)
+        >>> sc._expand_to_vector([60, 62], [2.0, 2.0])
```

Comment:
To make this more elegant you could consider making `_expand_to_vector` a class method and adding `step_contour_length` to its parameter list. Then you could write

```py
>>> StepContour._expand_to_vector([60, 62], [2.0, 2.0], step_contour_length=4)
[60, 60, 62, 62]
```

That I think would be preferable because you don't need to create a StepContour object to use the function.

Reply:
Added